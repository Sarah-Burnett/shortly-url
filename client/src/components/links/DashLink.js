import React from "react";
import styled from "styled-components";
import { grey, breakpoint } from "../Styles";
import DashLinkBtns from "./DashLinkBtns";
import { shortlyUrl } from "../../utilities/urls";
import {
  linkBackgroundColor as background,
  linkColor as color,
} from "../../utilities/themes";
import { useActiveLink } from "../../hooks/useActiveLink";

const Link = styled.div`
  cursor: pointer;
  max-width: 96vw;
  margin: 2vh 2vw;
  padding: 1vh 1vw;
  position: relative;
  overflow: hidden;
  @media (min-width: ${breakpoint}) {
    display: grid;
    grid-template-rows: auto auto;
    grid-template-columns: 1fr auto;
    grid-template-areas: "urls details" "buttons buttons";
  }
`;

const Urls = styled.div`
  grid-area: urls;
  position: relative;
  h3,
  input {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    width: 96vw;
    @media (min-width: ${breakpoint}) {
      width: 70vw;
    }
  }
`;

const Details = styled.div`
    grid-area: details;
    font-size: small;
    @media (min-width: ${breakpoint}) {
        text-align: right;
    }
     @media (max-width: ${breakpoint}){
        border-top: 1px solid ${grey};
        margin-top: 1vh;
        padding-top: 1vh;
    } 
    span {
        font-size: larger;
        white-space: nowrap;
    }
}
`;

const Input = styled.input`
  background: none;
  border: none;
  outline: none;
  font-family: inherit;
  font-size: inherit;
  color: inherit;
  min-width: 50vw;
`;

const DashLink = ({ link }) => {
  const { _id, date, full, click, short } = link;
  const { activeLink, setActiveLink } = useActiveLink();
  return (
    <Link
      style={{
        background: background(_id, activeLink),
        color: color(_id, activeLink),
      }}
      onClick={() => setActiveLink(link)}
    >
      <Urls className="urls">
        <h3>{full}</h3>
        <Input id={short} value={shortlyUrl + short} readOnly />
      </Urls>
      <Details className="details">
        <p>
          CREATED{" "}
          <span>{date.toString() && date.toString().substr(0, 10)}</span>
        </p>
        <p>
          CLICKS <span>{click.length}</span>
        </p>
      </Details>
      {_id === activeLink._id && <DashLinkBtns link={link} />}
    </Link>
  );
};

export default DashLink;
