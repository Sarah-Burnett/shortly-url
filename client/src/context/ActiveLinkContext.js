import React, { useState, createContext } from 'react';

export const ActiveLinkContext = createContext();

export const ActiveLinkProvider = (props) => {
    const [ activeLink, setActiveLink] = useState({
        activeLink: [],
        process: false
    });

    return (
        <ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
            {props.children}
        </ActiveLinkContext.Provider>
    )
};