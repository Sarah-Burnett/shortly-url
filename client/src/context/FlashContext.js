import React, { useState, createContext } from 'react';

export const FlashContext = createContext();

export const FlashProvider = (props) => {
    const [ flash, setFlash ] = useState({
        flashMsg: "",
        isFlashVisible: false,
        isAutoHideFlash: true,
    });
    return (
        <FlashContext.Provider value={[flash, setFlash]}>
            {props.children}
        </FlashContext.Provider>
    )
};