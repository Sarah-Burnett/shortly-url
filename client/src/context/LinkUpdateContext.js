import React, { useState, createContext } from 'react';

export const LinkUpdateContext = createContext();

export const LinkUpdateProvider = (props) => {
    const [ linkUpdate, setLinkUpdate] = useState();
    return (
        <LinkUpdateContext.Provider value={[linkUpdate, setLinkUpdate]}>
            {props.children}
        </LinkUpdateContext.Provider>
    )
};