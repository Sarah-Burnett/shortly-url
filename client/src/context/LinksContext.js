import React, { useState, createContext } from 'react';

export const LinksContext = createContext();

export const LinksProvider = (props) => {
    const [ links, setLinks] = useState([]);

    return (
        <LinksContext.Provider value={[links, setLinks]}>
            {props.children}
        </LinksContext.Provider>
    )
};