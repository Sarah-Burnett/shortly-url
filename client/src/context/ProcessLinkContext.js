import React, { useState, createContext } from 'react';

export const ProcessLinkContext = createContext();

export const ProcessLinkProvider = (props) => {
    const [ processLink, setProcessLink] = useState({
        link: [],
        process: false
    });
    return (
        <ProcessLinkContext.Provider value={[processLink, setProcessLink]}>
            {props.children}
        </ProcessLinkContext.Provider>
    )
};