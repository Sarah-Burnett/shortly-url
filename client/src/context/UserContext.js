import React, { useState, createContext } from 'react';

export const UserContext = createContext();

export const UserProvider = (props) => {
    const [ user, setUser] = useState({
        isAuth: false,
        firstName: undefined,
        id: undefined,
        token: undefined,
    });
    
    return (
        <UserContext.Provider value={[user, setUser]}>
            {props.children}
        </UserContext.Provider>
    )
};