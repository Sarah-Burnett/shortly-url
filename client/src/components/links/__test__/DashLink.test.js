import React, { useState } from "react";
import {
	rendersWithoutCrashing,
	matchesSnapshot,
	sampleLinks,
} from "../../../setupTests";
import DashLink from "../DashLink";
import { ActiveLinkContext } from "../../../context/ActiveLinkContext";
import { screen, fireEvent, render } from "@testing-library/react";
import { ProcessLinkProvider } from "../../../context/ProcessLinkContext";

const DashLinkWrapper = () => {
	const [activeLink, setActiveLink] = useState(sampleLinks[1]);
	return (
		<ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
			<ProcessLinkProvider>
				<DashLink link={sampleLinks[0]} />
			</ProcessLinkProvider>
		</ActiveLinkContext.Provider>
	);
};

it("renders dash link without crashing", () =>
	rendersWithoutCrashing(DashLinkWrapper));

it("dash link matches snapshot", () => matchesSnapshot(DashLinkWrapper));

it("dash link theme", async () => {
	render(<DashLinkWrapper />);
	expect(screen.getByTestId("link").style.background).toBe = "#FFF";
	await fireEvent.click(screen.getByTestId("link"));
	expect(screen.getByTestId("link").style.background).toBe =
		"hsl(257, 27%, 26%)";
});
