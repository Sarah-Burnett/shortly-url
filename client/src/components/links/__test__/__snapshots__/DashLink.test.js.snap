// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dash link matches snapshot 1`] = `
<div
  className="sc-fznyAO hycAEd"
  data-testid="link"
  onClick={[Function]}
  style={
    Object {
      "background": "#FFF",
      "color": "hsl(257, 27%, 26%)",
    }
  }
>
  <div
    className="sc-fznKkj ebYqPP urls"
  >
    <h3>
      http://reactcommunity.org/react-transition-group/css-transition
    </h3>
    <input
      className="sc-fzokOt jgLhP"
      id="octopus"
      readOnly={true}
      value="https://short.ly/octopus"
    />
  </div>
  <div
    className="sc-fznZeY jIxaEU details"
  >
    <p>
      CREATED
       
      <span>
        2020-08-14
      </span>
    </p>
    <p>
      CLICKS 
      <span>
        8
      </span>
    </p>
  </div>
</div>
`;
